docs/website/docs/dlt-ecosystem/verified-sources/pg_replication.md

```diff
@@ -0,0 +1,253 @@
+---
+title: Postgres replication
+description: dlt verified source for Postgres replication
+keywords: [postgres, postgres replication, database replication]
+---
+import Header from './_source-info-header.md';
+
+# Postgres replication
+
+<Header/>
+
+[Postgres](https://www.postgresql.org/) is one of the most popular relational database management systems. This verified source uses Postgres replication functionality to efficiently process tables (a process often referred to as *Change Data Capture* or CDC). It uses [logical decoding](https://www.postgresql.org/docs/current/logicaldecoding.html) and the standard built-in `pgoutput` [output plugin](https://www.postgresql.org/docs/current/logicaldecoding-output-plugin.html).
+
+Resources that can be loaded using this verified source are:
+
+| Name                 | Description                                     |
+| -------------------- | ----------------------------------------------- |
+| replication_resource | Load published messages from a replication slot |
+
+## Setup Guide
+
+### Setup user
+To setup a Postgres user follow these steps:
+
+1. The Postgres user needs to have the `LOGIN` and `REPLICATION` attributes assigned:
+    
+    ```sql
+    CREATE ROLE replication_user WITH LOGIN REPLICATION;
+    ```
+    
+2. It also needs `GRANT` privilege on the database:
+    
+    ```sql
+    GRANT CREATE ON DATABASE dlt_data TO replication_user;
+    ```
+    
+
+### Set up RDS
+To setup a Postgres user on RDS follow these steps:
+
+1. You must enable replication for RDS Postgres instance via [Parameter Group](https://docs.aws.amazon.com/AmazonRDS/latest/UserGuide/USER_PostgreSQL.Replication.ReadReplicas.html)
+
+2. `WITH LOGIN REPLICATION;` does not work on RDS, instead do:
+    
+    ```sql
+    GRANT rds_replication TO replication_user;
+    ```
+    
+3. Do not fallback to non-SSL connection by setting connection parameters:
+    
+   ```toml
+   sources.pg_replication.credentials="postgresql://loader:password@host.rds.amazonaws.com:5432/dlt_data?sslmode=require&connect_timeout=300"
+   ```
+### Initialize the verified source
+
+To get started with your data pipeline, follow these steps:
+
+1. Enter the following command:
+    
+   ```sh
+   dlt init pg_replication duckdb
+   ```
+    
+   It will initialize [the pipeline example](https://github.com/dlt-hub/verified-sources/blob/master/sources/pg_replication_pipeline.py) with a Postgres replication as the [source](https://dlthub.com/docs/general-usage/source) and [DuckDB](https://dlthub.com/docs/dlt-ecosystem/destinations/duckdb) as the [destination](https://dlthub.com/docs/dlt-ecosystem/destinations).
+    
+    
+2. If you'd like to use a different destination, simply replace `duckdb` with the name of your preferred [destination](https://dlthub.com/docs/dlt-ecosystem/destinations).
+    
+3. This source uses `sql_database` source, you can init it as follows:
```

Comment:
I narrowed down the dependency on `sql_database` in https://github.com/dlt-hub/verified-sources/pull/506. It's now only needed if a user does an initial load, i.e. when `persist_snaphots` is set to `True`. Good to mention that here.

Reply:
updated